### PR TITLE
sdk 1.2.0 bugs

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**
- remove contrast from theme options
- validate if parent option is null, if yes then set document.body

### CODE REVIEW

File: js/src/launchAsync.ts [hunk 1]
```diff
@@ -116,6 +116,7 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
+        options.parent = options.parent === null ? document.body : options.parent;
```
Comment (rwallerms) [inline js/src/launchAsync.ts:119]: checking for null is not the same as check for validity/existence. Am thinking along the lines of:

document.getElementById(parent)

to check if it really exists

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/src/launchAsync.ts (hunk 1 above)
Comment (cristobal-buenrostro) [reply js/src/launchAsync.ts]: done, 
parent is validated against undefined, if not undefined then check if existis in document, if not exists then set document.body

**Review by dylankil — APPROVED** [review]

File: js/src/launchAsync.ts [hunk 2]
```diff
@@ -116,7 +116,11 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
-        const parent = options.parent ? options.parent : document.body;
+        options.parent = typeof(options.parent) !== 'undefined' ? options.parent : document.body;
```
Comment (rwallerms) [inline js/src/launchAsync.ts:119]: still need to check for null, should use the way you were doing it before, since that is checking for both (truthy)

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/src/launchAsync.ts (hunk 2 above)
Comment (cristobal-buenrostro) [reply js/src/launchAsync.ts]: done, null and undefined are considered into parent validation

File: js/src/launchAsync.ts [hunk 3]
```diff
@@ -116,7 +116,11 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
-        const parent = options.parent ? options.parent : document.body;
+        options.parent = options.parent === null ? document.body : typeof(options.parent) !== 'undefined' ? options.parent : document.body;
```
Comment (rwallerms) [inline js/src/launchAsync.ts:119]: can't you just check null & undefined in one swoop by checking if truthy?

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/src/launchAsync.ts [hunk 4]
```diff
@@ -116,7 +116,11 @@ export function launchAsync(token: string, subdomain: string, content: Content,
...
+        if (!document.contains(options.parent)) {
```
Comment (rwallerms) [inline js/src/launchAsync.ts:120]: should combine this statement with above

const parent = options.parent && document.contains(options.parent) ? options.parent : document.body;

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/src/launchAsync.ts (hunk 3 above)
Comment (cristobal-buenrostro) [reply js/src/launchAsync.ts]: done

File: js/src/launchAsync.ts (hunk 4 above)
Comment (cristobal-buenrostro) [reply js/src/launchAsync.ts]: done as suggested by it is assigned to options.parent and then to parent variable, both are used later and need to have same value

File: js/src/launchAsync.ts [hunk 5]
```diff
@@ -116,7 +116,8 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
-        const parent = options.parent ? options.parent : document.body;
+        options.parent = options.parent && document.contains(options.parent) ? options.parent : document.body;
+        const parent = options.parent;
```
Comment (rwallerms) [inline js/src/launchAsync.ts:120]: assign const parent in the line above

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/src/launchAsync.ts (hunk 5 above)
Comment (cristobal-buenrostro) [reply js/src/launchAsync.ts]: done

**Review by rwallerms — APPROVED** [review]

**Pull request merged**